dashboard/lib/tasks/setup_or_migrate.rake

```diff
@@ -9,8 +10,10 @@ namespace :db do
     true
   end
 
-  task :setup_or_migrate => [:environment, :load_config] do
-    if database_exists?
+  task :setup_or_migrate do
+    db_exists = database_exists?
+    Rake::Task["db:load_config"].invoke
```

Comment:
Correct, the change was to preserve task ordering.

I quite simply don't know if `load_config` depends on `environment` or not, or whether it's order-sensitive (it's possible that `load_config` does not strictly depend on `environment`, but behaves differently if `environment` is invoked after it rather than before).

Because I don't know, in this PR I made an effort to match the exact task-dependency-ordering found in Rails's code for these two underlying tasks:
[`db:create`](https://github.com/rails/rails/blob/master/activerecord/lib/active_record/railties/databases.rake#L26)` => [:load_config]`
[`db:migrate`](https://github.com/rails/rails/blob/master/activerecord/lib/active_record/railties/databases.rake#L57)` => [:environment, :load_config]`

Here's what (I think) happens in this PR's `setup_or_migrate` task:
- Database doesn't exist:
  - `database_exists?` invokes `environment` and/or `Base.connection`, catches exception, returns `false` (the `:environment` task may or may not have successfully loaded at this point, depending on whether the specific environment's initializers attempted to establish a database connection or not.)
  - invokes `load_config`
  - invokes `db:create`
- Database does exist:
  - `database_exists?` invokes `environment` and/or `Base.connection`, establishes connection, returns `true`
  - invokes `load_config`
  - invokes `db:migrate`

The assumption was that if Rails specifically constructed its task-dependency-ordering like this, it was possibly (probably?) for a reason, and I wanted to introduce as little modified behavior as possible.